File: hermes/src/main/java/org/eclipse/paho/android/service/contract/UnsubscribeMqttEventActionListener.java
package org.eclipse.paho.android.service.contract;

import org.eclipse.paho.android.service.event.MqttAction;
import org.eclipse.paho.android.service.event.MqttEvent;
import org.eclipse.paho.client.mqttv3.IMqttToken;
import org.greenrobot.eventbus.EventBus;

/**
 * 默认实现MQTT消息监听接口
 * @author zhouL
 * @date 2023/10/9
 */
public class UnsubscribeMqttEventActionListener extends DefMqttEventActionListener {
    private String[] topics;


    public UnsubscribeMqttEventActionListener(String clientKey, String errorMsg, String[] topics) {
        super(clientKey, MqttAction.ACTION_UNSUBSCRIBE, errorMsg);
        this.topics = topics;
    }

    @Override
    public void onSuccess(IMqttToken asyncActionToken) {
        EventBus.getDefault().post(MqttEvent.createUnsubscribeSuccess(clientKey, topics));
    }

    @Override
    public void onFailure(IMqttToken asyncActionToken, Throwable exception) {
        exception.printStackTrace();
        EventBus.getDefault().post(MqttEvent.createUnsubscribeFail(clientKey, topics, errorMsg, exception));
    }
}

File: hermes/src/main/java/com/lodz/android/hermes/mqtt/base/contract/UnsubscribeMqttEventActionListener.kt
package com.lodz.android.hermes.mqtt.base.contract

import com.lodz.android.hermes.mqtt.base.bean.eun.MqttAction
import com.lodz.android.hermes.mqtt.base.bean.event.MqttEvent
import org.eclipse.paho.client.mqttv3.IMqttToken
import org.greenrobot.eventbus.EventBus

/**
 * 解订阅监听接口实现
 * @author zhouL
 * @date 2023/10/12
 */
class UnsubscribeMqttEventActionListener(
    clientKey: String,
    errorMsg: String,
    private val topics: Array<String>
) : DefMqttEventActionListener(clientKey, MqttAction.ACTION_UNSUBSCRIBE, errorMsg) {

    override fun onSuccess(asyncActionToken: IMqttToken?) {
        EventBus.getDefault().post(MqttEvent.createUnsubscribeSuccess(clientKey, topics))
    }

    override fun onFailure(asyncActionToken: IMqttToken?, exception: Throwable?) {
        exception?.printStackTrace()
        EventBus.getDefault().post(MqttEvent.createUnsubscribeFail(clientKey, topics, errorMsg, getThrowable(exception)))
    }
}